test/plugins.js

```diff
@@ -1,42 +1,110 @@
-import * as chai from '../index.js';
+import {use, assert, expect, Should} from '../index.js';
+
+/**
+ * A chai plugin that adds the `testing` property on chai assertions.
+ *
+ * @param {unknown} chai
+ */
+function plugin(chai) {
+  if (chai.Assertion.prototype.testing) return;
+
+  chai.assert.testing = 'successful';
+
+  Object.defineProperty(chai.Assertion.prototype, 'testing', {
+    get: function () {
+      return 'successful';
+    },
+  });
```

Comment:
I thought it was surprising I had to do this. Unless I'm misunderstanding how this is supposed to work.

If it's correct that we need to define a property on both `chai.assert` and `chai.Assertion.prototype` I was thinking we could think about a more user friendly way to write plugins for the next breaking version.

Reply:
you would normally use `util.addProperty` at least, like:

```ts
util.addProperty(Assertion.prototype, 'testing', function () {
  this.assert(this._obj === 'foo');
});
```

so expect and should will inherit that

but you're right you'd still have to manually extend the assert interface i think, something like:

```ts
assert.isTesting = function() {
  new Assertion(val).testing;
}
```